video/out/vo_gpu_next.c

```diff
@@ -1612,6 +1528,61 @@ static char *get_cache_file(struct priv *p)
     return file;
 }
 
+static void load_cache_files(struct priv *p)
+{
+    char *icc_cache = get_cache_file(p, "icc");
+    char *shader_cache = get_cache_file(p, "shader");
+    bool same_cache = strcmp(icc_cache, shader_cache) == 0;
+    if (shader_cache) {
+        FILE *cache = fopen(shader_cache, "rb");
+        if (cache) {
+            pl_cache_load_file(p->shader_cache, cache);
+            if (same_cache)
+                pl_cache_load_file(p->icc_cache, cache);
+            fclose(cache);
+        }
+        talloc_free(shader_cache);
+    }
+    if (icc_cache) {
+        if (!same_cache) {
+            FILE *cache = fopen(icc_cache, "rb");
+            if (cache) {
+                pl_cache_load_file(p->icc_cache, cache);
+                fclose(cache);
+            }
+        }
+        talloc_free(icc_cache);
+    }
+}
+
+static void save_cache_files(struct priv *p)
+{
+    char *icc_cache = get_cache_file(p, "icc");
+    char *shader_cache = get_cache_file(p, "shader");
+    bool same_cache = strcmp(icc_cache, shader_cache) == 0;
+    if (shader_cache) {
+        FILE *cache = fopen(shader_cache, "wb");
+        if (cache) {
+            pl_cache_save_file(p->shader_cache, cache);
+            if (same_cache)
+                pl_cache_save_file(p->icc_cache, cache);
+            fclose(cache);
+        }
+        talloc_free(shader_cache);
+    }
+    if (icc_cache) {
+        if (!same_cache) {
+            FILE *cache = fopen(icc_cache, "wb");
+            if (cache) {
+                pl_cache_save_file(p->icc_cache, cache);
+                fclose(cache);
+            }
+        }
+        talloc_free(icc_cache);
+    }
+}
```

Comment:
Maybe add some error logging here? "failed to load cache" "failed to save cache" and so on. Maybe even trace ones when  it loads successfully with size.

Reply:
Well libplacebo already logs successful saved/loaded objects, but yes we can log a warning if it fails.

Reply:
At your discretion. I noticed a lot of I/O without logging, that generally don't end well.

Reply:
My log messages are not very creative but I tried something.